src/commands/lightning/dev/component.ts

```diff
@@ -63,48 +87,49 @@ export default class LightningDevComponent extends SfCommand<void> {
             return undefined;
           }
 
-          const componentName = path.basename(componentPath);
-          const label = ComponentUtils.componentNameToTitleCase(componentName);
+          const name = path.basename(componentPath);
+          const label = ComponentUtils.componentNameToTitleCase(name);
 
           return {
-            name: componentName,
+            name,
             label: xml.LightningComponentBundle.masterLabel ?? label,
             description: xml.LightningComponentBundle.description ?? '',
           };
         })
       )
     ).filter((component) => !!component);
 
-    let name = flags.name;
-    if (!flags['client-select']) {
-      if (name) {
+    if (!clientSelect) {
+      if (componentName) {
         // validate that the component exists before launching the server
-        const match = components.find((component) => name === component.name || name === component.label);
+        const match = components.find(
+          (component) => componentName === component.name || componentName === component.label
+        );
         if (!match) {
-          throw new Error(messages.getMessage('error.component-not-found', [name]));
+          throw new Error(messages.getMessage('error.component-not-found', [componentName]));
         }
 
-        name = match.name;
+        componentName = match.name;
       } else {
         // prompt the user for a name if one was not provided
-        name = await PromptUtils.promptUserToSelectComponent(components);
-        if (!name) {
+        componentName = await PromptUtils.promptUserToSelectComponent(components);
+        if (!componentName) {
           throw new Error(messages.getMessage('error.component'));
         }
       }
     }
 
-    const dirname = path.dirname(url.fileURLToPath(import.meta.url));
-    const rootDir = path.resolve(dirname, '../../../..');
-    const port = parseInt(process.env.PORT ?? '3000', 10);
-
-    await cmpDev({
-      rootDir,
-      mode: 'dev',
-      port,
-      name: name ? `c/${name}` : undefined,
-      namespacePaths,
-      open: process.env.OPEN_BROWSER === 'false' ? false : true,
-    });
+    await startLWCServer(logger, sfdxProjectRootPath, ldpServerToken, Platform.desktop, serverPorts);
+
+    const targetOrgArg = PreviewUtils.getTargetOrgFromArguments(this.argv);
+    const launchArguments = PreviewUtils.generateComponentPreviewLaunchArguments(
+      ldpServerUrl,
+      ldpServerId,
+      componentName,
+      targetOrgArg
+    );
+
+    // Open the browser and navigate to the right page
+    await this.config.runCommand('org:open', launchArguments);
```

Comment:
@bpbuch I need to be able to get the proper URL and iframe it inside of the VSCode webview. Right now it works because I know the URL is localhost, but with this change how will I be able to get the right URL and set that into the iframe?

Reply:
Also we still want a way to not automatically launch the browser when running the command in VSCode / Code Builder (i.e. we only want to open the preview panel)